### DISCUSSION TITLE
Creation of runtime packages

### DISCUSSION THREAD

**Discussion by frottke** [microsoft/AL-Go#1574] [Ideas]
As @nixpt82 asked in #486, is there a way to generate runtime packages as an app file? We have a product that we want to distribute onprem, but we have to generate the runtime packages manually, which is quite tedious as you can imagine.

If there is no solution within Al-Go, do you have a PoC for overriding the default behavior? It seems that there is "GetBcContainerAppRuntimePackage.ps1", but it doesn't seem to be executed in the GitHub action "CI/CD"?

**Comment by freddydk** [comment]
Currently there is not - but... - we are working on this from multiple angles.
Cannot give you any dates yet - but allow me to get back after I get back from Directions EMEA

**Comment by larswestman** [comment]
@frottke I used this as inspiration for a powershell script that create runtime packages: https://al-go.dev/releasing-on-premises-extensions/ Works fine but takes a lot of time to run. It would be great to have it built in in AL-Go. I'm currently developing on BC20.0 and there's a lot of majors/minors to create runtime packages for since BC20.0....

**Comment by freddydk** [comment]
I am working on this now - just signed up for doing a Directions Webinar in January and expect to have something to show there.

**Comment by frottke** [comment]
Are there any news regarding this issue?

**Comment by PropmanProphecy** [comment]
@freddydk  Is there any update on this issue?

**Comment by ralf-escher** [comment]
@freddydk we would also like to generate runtime packages and are waiting for updates

**Comment by naviprobeha** [comment]
Any news here? Would really appriciate it.

**Comment by jonaswre** [comment]
Just going to drop this here. 
https://github.com/BusinessCentralApps/GenerateBcNuGet

Its still a POC and not very rounded but we are using it.
